pkg/kubectl/cmd/util/editor/editoptions.go

```diff
@@ -336,6 +364,87 @@ func (o *EditOptions) Run() error {
 	}
 }
 
+func (o *EditOptions) applyEditPatch(oringalInfos, infos []*resource.Info) error {
+	for ix := range infos {
+		updatedInfo := infos[ix]
+
+		originalJS, err := encodeToJson(o.Encoder, oringalInfos[ix].Object)
+		if err != nil {
+			return err
+		}
+
+		editedJS, err := encodeToJson(o.Encoder, updatedInfo.Object)
+		if err != nil {
+			return err
+		}
+
+		if reflect.DeepEqual(originalJS, editedJS) {
+			cmdutil.PrintSuccess(o.Mapper, false, o.Out, updatedInfo.Mapping.Resource, updatedInfo.Name, false, "skipped")
+		} else {
+			o.annotationPatch(updatedInfo)
+			cmdutil.PrintSuccess(o.Mapper, false, o.Out, updatedInfo.Mapping.Resource, updatedInfo.Name, false, "edited")
+		}
+	}
+	return nil
+}
+
+func (o *EditOptions) annotationPatch(update *resource.Info) error {
+	patch, _, err := GetApplyPatch(update, o.Encoder, false)
+	if err != nil {
+		return err
+	}
+	mapping := update.ResourceMapping()
+	client, err := o.f.UnstructuredClientForMapping(mapping)
+	if err != nil {
+		return err
+	}
+	helper := resource.NewHelper(client, mapping)
+	_, err = helper.Patch(o.CmdNamespace, update.Name, types.StrategicMergePatchType, patch)
+	if err != nil {
+		return preservedFile(err, o.file, o.ErrOut)
```

Comment:
if you need the filename here, plumb it as an arg into `applyEditPatch` and `annotationPatch`, though I'm confused about why we need the file here... returning an error from annotationPatch should propagate up and return an error from applyEditPatch, which should trigger returning a preservedFile error